src/entity/Company.java

```diff
@@ -11,9 +14,10 @@
 public class Company {
     private static Company company;
     private static AtomicBoolean instanceCreated = new AtomicBoolean(false);
+    private static AtomicBoolean carsCreated = new AtomicBoolean(false);
```

Comment:
You don't need additional check of the cars creation. You will initialize cars only once after new Company(). This is thread-safely.